### DISCUSSION TITLE
FleetDM API Only user setup

### DISCUSSION THREAD

**Discussion by adamphetamine** [Security-Onion-Solutions/securityonion#8884] [Unsupported Versions] [answered]
Hi,
I have logged in to fleetctl and set up an API only user. 
According to this doc
[https://fleetdm.com/docs/using-fleet/fleetctl-cli](https://fleetdm.com/docs/using-fleet/fleetctl-cli)

I should then be able to use curl or Postman to get the permanent API Token by hitting the endpoint 

POST /api/v1/fleet/login

But I cannot get this to respond. The convention with SO seems to be adding container name to the URL, but even hitting

https://domain.com/fleet/api/v1/fleet/login

Gives no joy- timeout. What am I doing wrong please?

**Comment by adamphetamine** [comment]
maybe I should re-phrase this so I'm not wasting my time if this is impossible-

Does anyone have an API user working for FleetDM in Security Onion?

**Comment by defensivedepth** [comment, marked as answer]
I have not tested the API-only user, however, it should work.

My `Go To` method when figuring stuff like this out is to pop open my browser's dev tools and watch the network requests as I navigate through the app - for instance:

![image](https://user-images.githubusercontent.com/954732/195336648-207a3185-5859-479e-a87e-476b63b61706.png)


Taking that URL & curl'ing it with my API key:

![image](https://user-images.githubusercontent.com/954732/195338147-9325bd57-d762-4947-ac8f-28f1a726819a.png)

**Comment by adamphetamine** [comment]
ah good idea, thanks- will report back if I figure it out, thanks again

**Comment by adamphetamine** [comment]
ok I did find this-

https://defensivedepth.com/2020/04/02/kolide-fleet-breaking-out-the-osquery-api-web-ui/

Which basically notes
- calls to the web UI API in Fleet are split out from calls to the OSQuery API

So it looks like incoming API calls to port 8090 will be dropped as they aren't gRPC
I think this means my API user calls will not connect because I need to hit the OSQuery endpoint with a user that's in the Web UI context.
I'm sorry I can't confirm this by reading the nginx config as I'm not familiar with it, but it does look very similar to the code in the blog post for 2 years ago.

Right now, this seems to be a feature not a bug, so I'll try to get the data another way, thanks.

**Comment by adamphetamine** [comment]
ok I figured it out several months later-

It's not an issue with either the nginx proxy server or the iptables firewall- the issue for me was using the wrong API endpoint 

FleetDM docs say to use

https://domain.com/fleet/api/v1/fleet/login

but for Security Onion the endpoint is

https://domain.com/fleet/api/latest/fleet/login

I hope this helps someone in the future
